src/lyra.ts

```diff
@@ -326,7 +407,7 @@ export async function insertBatch<S extends PropertiesSchema>(
 
       for (const line of batch) {
         try {
-          insert(lyra, line, config);
+          await insertWithHooks(lyra, line, config);
```

Comment:
As you can see, `insertBatch` returns a `Promise<void>`. Maybe we want to return a list of IDs or something? What would be the benefit of using hooks here otherwise? wdyt?

Reply:
The advantage is that you can control when and what to do after each insertion, for example. But, since the `insertBatch` signature didn't change, that's sounds a reasonable change.

I'd not return the list of IDs, since we have the hooks they can use it if they want, otherwise, we'll need to increase the memory consumption to do that.

Reply:
Hey @micheleriva @RafaelGSS  can we return a list of ids from insertBatch. There is no option for a custom id, so insertBatch is not of use if you want to maintain a mapping with your database for invalidation of data.

Reply:
If all the create/update functions return id's then insertBatch  should also return id's preserving order

Reply:
The memory argument is valid, maybe if not add return of id's here then add something in the documentation pointing towards `insertWithHooks` function when you need to get list of id

Reply:
This will be resolved when we rehaul the hook system in the future with more hooks. For now I agree not to increase memory consumption.

Reply:
@ShogunPanda I am not sure if the new hooks system has been added or not, but what do you think about my point of adding something in the documentation pointing towards `insertWithHooks` when you need to get a list of id? 

I was stuck again today on the same thing with my old code and was reminded of this!

Reply:
@ravaan insertWithHooks does not exist anymore. Nowadays you can provide `afterInsert 
` when doing the `create` to achieve the same result.

See: https://docs.oramasearch.com/internals/components#afterinsert

Does it answer your question?